Rakefile

```diff
@@ -1 +1,95 @@
-require "bundler/gem_tasks"
+require 'bundler/gem_tasks'
+require 'rake/clean'
+require 'cucumber/rake/task'
+require 'mechanize'
+require 'fileutils'
+require 'yaml'
+
+CDK_DOWNLOAD_URL='https://access.redhat.com/downloads/content/293/ver=2/rhel---7/2.0.0/x86_64/product-software'
+CDK_BOX_BASE_NAME='rhel-cdk-kubernetes-7.2-23.x86_64.vagrant'
+
+ADB_DOWNLOAD_URL='http://cloud.centos.org/centos/7/atomic/images'
+ADB_BOX_BASE_NAME='AtomicDeveloperBundle-2.0.0-CentOS7'
+
+CLOBBER.include('pkg')
+CLEAN.include('build')
+
+Cucumber::Rake::Task.new(:features)
+
+namespace :features do
+  cucumber_config = YAML.load_file('./.config/cucumber.yml')
+  cucumber_config.each_key do |key|
+    if key == 'default'
+      next
+    end
+    Cucumber::Rake::Task.new(key.to_sym, "Run Cucumber features using profile '#{key}' (see cucumber.yml)'") do |t|
+      t.profile = key
+    end
+  end
+end
+
+desc 'Download CDK Vagrant box using the specified provider (default \'virtualbox\')'
+task :get_cdk, [:provider] do |t, args|
+  if args[:provider].nil?
+    provider='virtualbox'
+  else
+    provider=args[:provider]
+  end
+  agent = Mechanize.new
```

Comment:
I would prefer to take out these lower level operations in separate file and then call it here.

Adding these code here actually make the `Rakefile` bulky and would be difficult to read it as a whole when it scales up.